src/fetch/contracts.py

```diff
@@ -98,12 +101,47 @@ def results(self, load_from: NetworkFile) -> dict[str, bool]:
             partition = addresses[index:index + self.max_batch_size]
             results |= self._limited_is_contract(partition)
 
-        confirmed_contracts = sorted([k for k, v in results.items() if v])
+        confirmed_contracts = sorted([Account(k) for k, v in results.items() if v])
         print(f"found {len(confirmed_contracts)} contracts, writing to file")
-        with open(load_file.filename(), 'w', encoding='utf-8') as file:
-            file.write('\n'.join(confirmed_contracts))
+        write_to_csv(data_list=confirmed_contracts, outfile=load_file)
         return results
 
+    def _limited_balances(self, addresses: list[str]) -> dict[str, int]:
+        request_data = []
+        for index, eth_address in enumerate(addresses):
+            request_data.append({
+                "jsonrpc": "2.0",
+                "method": "eth_getBalance",
+                "params": [eth_address, "latest"],
+                "id": index
+            })
+        response = requests.post(self.node_url, json=request_data)
```

Comment:
Is this how batch requests are sent to a node? By sending a vector of requests? This is great! I like how things are handled.